src/Airship/RST.hs

```diff
@@ -0,0 +1,137 @@
+{-# LANGUAGE AllowAmbiguousTypes   #-}
+{-# LANGUAGE BangPatterns          #-}
+{-# LANGUAGE FlexibleInstances     #-}
+{-# LANGUAGE MultiParamTypeClasses #-}
+{-# LANGUAGE TypeFamilies          #-}
+{-# LANGUAGE UndecidableInstances  #-}
+
+{-
+  RST is like the RWST monad, but has no Writer instance, as Writer leaks space.
+  This file is almost entirely lifted from the Snap framework's implementation.
+-}
+
```

Comment:
Should we include the Snap copyright and license here, to indicate this file is specifically the one we're giving proper attribution?